Automattic/wp-calypso
# Move Email Management to `/email` route

Some users wish to buy or manage G Suite without necessarily buying or managing their domain on WordPress.com. Currently G Suite is only accessible via (and nested under) the Domains management screens and paths, i.e. `/domains/manage/email/foo.com`. 

We also check that you have a domain when a user tries to visit any of the email management screens directly.

This patch moves email management to its own top-level `/email/foo.com` path and allows it to work without a domain. No other functionality should be affected.

#### Changes

* Email is now managed from `/email` routes. A mostly complete list of these changes are below:
  * `domains/manage/email` *->* `/email`
  * `domains/manage/email/:site` *->* `/email/:site`
  * `domains/manage/add-gsuite-users/:site` ->`email/add-gsuite-users/:site`
  * `domains/manage/:domainName/add-gsuite-users/:site` ->`email/:domainName/add-gsuite-users/:site`
  * `domains/manage/email/:domainName/edit/:site` *->* `/email/:domainName/manage/:site`

* Email has been separated from the Domains Section and given it's own Email Section
* Miscellaneous fixes to CSS names and opportunistic updates from `google-apps` -> `gsuite`

#### Testing Instructions

*Prerequisites*
* A site with a domain and GSuite
* A site with a domain and without GSuite ( can be the same as the above site )  
* A site without a domain

*Testing New Routes*
1. Navigate to `/email/:site` for your site with GSuite
1. Confirm that the page is unchanged ( see screenshot below for reference, though note you may or may not have the same notices ) <img width="1065" alt="screen shot 2019-02-19 at 2 36 00 pm" src="https://user-images.githubusercontent.com/2810519/53052521-d27f5180-3453-11e9-8847-c7139f63f71b.png">
1. Click one of the "Add GSuite Users" Button
1. Observe the new URL. Confirm it matches `/email/:domain/add-gsuite-users/:site`
1. Observe the page and confirm it is unchanged ( screenshot below ) <img width="739" alt="screen shot 2019-02-19 at 3 37 09 pm" src="https://user-images.githubusercontent.com/2810519/53055356-3dcd2180-345c-11e9-9108-e98b0da9db84.png">
1. Click the "Back" Button
1. Observe the new URL. Confirm that it matches `/email/:domain/manage/:site`
1. Navigate to `/email/:domain/forwarding/:site` for the site with a domain and without g Suite
1. Confirm the Email Forwarding page is uncharged ( screenshot below ) <img width="741" alt="screen shot 2019-02-19 at 3 39 55 pm" src="https://user-images.githubusercontent.com/2810519/53055451-9e5c5e80-345c-11e9-9a5b-189245c54a0f.png">
1. Test Email Forwarding with the following Instructions
   1. Add email forward
   1. Observe that email forward appears at top
   1. Click mail icon to resend verification email
   1. Verify forward in your email
   1. Refresh, observe that email icon is gone
   1. Click delete icon
   1. Observe that email forward disappear
1. Click the Back Button
1. Observe the new URL. Confirm that it matches `/email/:domain/manage/:site`
1. Confirm that you are shown the G Suite CTA and that it is unchanged  <img width="1059" alt="screen shot 2019-02-19 at 3 41 31 pm" src="https://user-images.githubusercontent.com/2810519/53055645-4f62f900-345d-11e9-8555-5a750e2917f1.png">
1. Click "Add G Suite"
1. Confirm you are taken to the add G Suite users page again
1. Fill in the info and click continue
1. Confirm G Suite was correctly added to your cart

*Testing Legacy Redirects*
1. Navigate to `/domains/manage/email`
1. Confirm you are taken to `/email` and given a list of sites ( if you have more than one)
1. Navigate to `/domains/manage/email/:site` for your site with G Suite
1. Confirm you are taken to `/email/:site`
1. Navigate to `/domains/manage/email/:domain/edit/:site`
1. Confirm you are taken to `/email/:domain/manage/:site`
1. Navigate to `/domains/manage/:domain/email-forwarding/:site
1. Confirm you are taken to `/email/:domain/forwarding/:site`
1. Navigate to `/domains/manage/add-gsuite-users/:site`
1. Confirm you are taken to `/email/:site`
1. Navigate to `/domains/manage/:domain/add-gsuite-users/:site`
1. Confirm you are taken to `/email/:domain/add-gsuite-users/:site`
